src/lib/orion_websocket/parser.cpp

```diff
@@ -56,4 +65,19 @@ void ws_parser_parse(std::string &url, std::string &verb, std::string &payload,
   }
   else
     payload.clear();
+
+  if (doc.HasMember("headers"))
+  {
+    std::map<std::string, std::string *>::iterator it = head.supportedHeader.begin();
+    while (it != head.supportedHeader.end())
+    {
+      const char *value = doc["headers"][it->first.c_str()].GetString();
+      *(it->second) = value ? std::string(value) : std::string();
+      ++it;
+    }
+
+    head.gotHeaders = true;
+  }
+  else
+    head.gotHeaders = false;
```

Comment:
Another style rule we have (Fermins idea), is to always use brackets '{}', even with one line blocks.